gobblin-runtime/src/test/java/org/apache/gobblin/spec_catalog/FlowCatalogTest.java

```diff
@@ -117,6 +127,45 @@ public void cleanUp() throws Exception {
     }
   }
 
+  /**
+   * Make sure that when there's on spec failed to be deserialized, the rest of spec in specStore can
+   * still be taken care of.
+   */
+  @Test
+  public void testGetSpecRobustness() throws Exception {
```

Comment:
Can the test include reading a 0 byte spec from the specStore? Also, can the readSpecFromFile() method in TestSpecStore call SerializationUtils.deserialize() on the 0 byte file? This way we can have it throw a SerializationException.